infra/modules/gcp/main.tf

```diff
@@ -26,6 +27,36 @@ resource "google_project_service" "gcp_infra_api" {
   disable_on_destroy         = false # disabling on destroy has potential to conflict with other uses of the project
 }
 
+resource "google_artifact_registry_repository" "psoxy-functions-repo" {
```

Comment:
Actually this is *not* required, as if we don't specify the repo GCP will create a new one:

![image](https://github.com/Worklytics/psoxy/assets/75276364/20ad87fa-dec8-4c6c-8db4-fda6740c1bfa)

But I think it is cleaner if we create one for psoxy images, so we could customize some data there...

Reply:
can we make it conditional or something?

Otherwise it's a breaking change, as technically we have the version for google provider in examples at       `version = ">= 3.74, <= 5.0"` -- and this is still in beta < 4.0